git/client.go

```diff
@@ -266,6 +266,15 @@ func (c *Client) CreateBranch(name string, commit Commitish) error {
 		return err
 	}
 
+	if name == "HEAD" {
+		return fmt.Errorf("fatal: 'HEAD' is not a valid branch name.")
+	}
+
+	// Create the file so that any parent directories are created
```

Comment:
I'm not sure that I understand this sentence. I think it's trying to imply that `Create` creates parents directories while the `WriteFile` below doesn't, but I'm not even sure if that's true (`Create` is a thin wrapper around `os.Create`, and `os.Create`'s doesn't say anything about what happens if the parent directory(s) aren't present)

Reply:
This Create() will do an os.MkdirAll() creating any necessary parent directories while WriteFile doesn't. Should I fix WriteFile so that it does?

Reply:
nevermind, I was looking at GitDir.Create (which doesn't do a MkdirAll), not File.Create (which does), so the code makes sense.

I still think the wording of the comment can be cleaned up a bit. How about something like:
```suggestion
	// Create the file using File.Create first to ensure any parent directories are created
	// before writing the file contents.
```